jetty-server/src/test/java/org/eclipse/jetty/server/HttpOutputInterceptorTest.java

```diff
@@ -0,0 +1,525 @@
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.HttpTester;
+import org.eclipse.jetty.logging.StacklessLogging;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+import org.eclipse.jetty.server.handler.HandlerCollection;
+import org.eclipse.jetty.util.Callback;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class HttpOutputInterceptorTest
+{
+    private static final Logger LOG = LoggerFactory.getLogger(HttpOutputInterceptorTest.class);
+
+    private Server _server;
+    private LocalConnector _localConnector;
+    private HandlerCollection _handlerCollection;
+    private StacklessLogging _stacklessLogging;
+
+    @BeforeEach
+    public void before() throws Exception
+    {
+        _server = new Server();
+        HttpConfiguration config = new HttpConfiguration();
+        config.setOutputBufferSize(1024);
+        config.setOutputAggregationSize(256);
+
+        _localConnector = new LocalConnector(_server, new HttpConnectionFactory(config));
+        _localConnector.setIdleTimeout(Duration.ofMinutes(1).toMillis());
+        _server.addConnector(_localConnector);
+        ServerConnector serverConnector = new ServerConnector(_server, new HttpConnectionFactory(config));
+        _server.addConnector(serverConnector);
+
+        _handlerCollection = new HandlerCollection();
+        _server.setHandler(_handlerCollection);
+        _stacklessLogging = new StacklessLogging(HttpChannel.class);
+    }
+
+    @AfterEach
+    public void after() throws Exception
+    {
+        _stacklessLogging.close();
+        _server.stop();
+    }
+
+    @Test
+    public void testInterceptorCallbackFailed() throws Exception
+    {
+        addInterceptor(nextInterceptor -> new HttpOutput.Interceptor()
+        {
+            @Override
+            public void write(ByteBuffer content, boolean last, Callback callback)
+            {
+                callback.failed(new RuntimeException("callback failed from interceptor"));
+            }
+
+            @Override
+            public HttpOutput.Interceptor getNextInterceptor()
+            {
+                return nextInterceptor;
+            }
+        });
+
+        CompletableFuture<Throwable> errorFuture = new CompletableFuture<>();
+        _handlerCollection.addHandler(new AbstractHandler()
+        {
+            @Override
+            public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException
+            {
+                baseRequest.setHandled(true);
+                byte[] content = "content to be written".getBytes();
```

Comment:
Always specify the charset. Unnecessary here, but for good measure and consistency with rest of code.